src/apps/VirtualApp/components/Monitoring.js

```diff
@@ -0,0 +1,79 @@
+import React, {
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
+import {
+  Icon,
+  Menu,
+  Placeholder,
+  Popup,
+  Table,
+} from 'semantic-ui-react';
+
+const usePrevious = (value) => {
+  const ref = useRef();
+  useEffect(() => {
```

Comment:
What do you need useEffect for?

Reply:
This hook is just to get the previous value of a prop it is being used for.
See line 26